src/reducers/entities.js

```diff
@@ -13,24 +13,24 @@ const reducer = handleActions({
     [action.payload.entityType]: {
       ...state[action.payload.entityType],
       [action.payload.data.id]: {
-        ...get(state, `[${action.payload.entityType}][${action.payload.data.id}]`, {}),
+        ...get(Object.assign({}, state), `[${action.payload.entityType}][${action.payload.data.id}]`, {}),
         ...action.payload.data,
       },
     },
   }),
   [receiveEntities]: (state, action) => {
-    const result = { ...state }
+    const result = { ...(Object.assign({}, state)) }
     keys(action.payload.entities).forEach((entityType) => {
       // preserve old entities
-      const entitiesState = get(state, `${entityType}`, {})
+      const entitiesState = { ...(get(state, `${entityType}`, {})) }
       set(result, `${entityType}`, entitiesState)
 
       keys(action.payload.entities[entityType]).forEach((entityId) => {
```

Comment:
Why you just not build an object of entities and once you have finished iterating over them, you just include that new object to the state? In that way, you would save time and resources because if for any reason the state is too big and has other stuff not related with this "reducer" you are copying it in different places, right?